app/models/course.rb

```diff
@@ -141,26 +142,26 @@ def initialize_duplicate(duplicator, other)
     self.creator = duplicator.options[:current_user]
     self.registration_key = nil
     logo.duplicate_from(other.logo) if other.logo_url
+    material_folders << duplicator.duplicate(other.root_folder)
+  end
 
-    # This also duplicates assessments.
-    self.assessment_categories = duplicator.duplicate(other.assessment_categories)
-    self.lesson_plan_items = duplicator.duplicate(other.lesson_plan_items.map(&:actable)).
-                             map(&:acting_as)
-    self.lesson_plan_milestones = duplicator.duplicate(other.lesson_plan_milestones)
-
-    # Find concrete material_folders and only duplicate those.
-    # This must be done after duplicating assessments.
-    # Do not try duplicating all folders at once. Parent IDs do not seem to be populated by the
-    # edge gem until the database entries are created.
-    material_folders_to_duplicate = other.material_folders.concrete
-    self.material_folders = duplicator.duplicate(material_folders_to_duplicate).compact
-
-    # Skill branches are duplicated as part of skills.
-    self.assessment_skills = duplicator.duplicate(other.assessment_skills)
-
-    self.levels = duplicator.duplicate(other.levels)
-    self.achievements = duplicator.duplicate(other.achievements)
-    self.forums = duplicator.duplicate(other.forums)
+  # List of top-level items that need to be duplicated for the whole course to be considered duplicated.
+  def duplication_manifest
```

Comment:
This manifest still instantiates all the objects, which could take a large amount of memory.  Can we use a way to only instantiate the objects which are currently being duplicated?  That means it should not be referenced after save, thus can be garbage collected.